test/SolverTrampoline.ts

```diff
@@ -55,6 +85,83 @@ describe("SolverTrampoline", function () {
     });
   });
 
+  describe("settle", function () {
+    it("Should execute a settlement and increment nonce", async function () {
+      const { solverTrampoline, signTestSettlement, solver } =
+        await loadFixture(fixture);
+
+      const nonce = await solverTrampoline.nonces(solver.address);
+      const {
+        settlement,
+        r,
+        s,
+        v,
+      } = await signTestSettlement(solver, nonce);
+
+      await expect(solverTrampoline.settle(settlement, nonce, r, s, v))
+        .to.not.be.reverted;
+      expect(await solverTrampoline.nonces(solver.address))
+        .to.equal(nonce.add(1));
+    });
+
+    it("Should propagate settlement reverts", async function () {
+      const { solverTrampoline, signTestSettlement, solver } =
+        await loadFixture(fixture);
+
+      const nonce = await solverTrampoline.nonces(solver.address);
+      const {
+        settlement,
+        r,
+        s,
+        v,
+      } = await signTestSettlement(solver, nonce, { shouldRevert: true });
+
+      await expect(solverTrampoline.settle(settlement, nonce, r, s, v))
+        .to.be.revertedWith("test settlement reverted");
+    });
+
+    it("Should deny settlements with invalid signatures", async function () {
+      const { solverTrampoline } = await loadFixture(fixture);
+
+      const { r, s, v } = INVALID_SIGNATURE;
+      await expect(solverTrampoline.settle("0x", 0, r, s, v))
+        .to.be.reverted;
```

Comment:
Why not checking that the error is `Unauthorized`? It [looks like](https://hardhat.org/hardhat-chai-matchers/docs/overview#reverts) they finally introduced `revertedWithCustomError`! (Same for other revert statements in the tests.)

Reply:
Nice! I didn't know, will update the tests.